components/user-store/org.wso2.carbon.identity.user.store.count/src/main/java/org/wso2/carbon/identity/user/store/count/UserStoreCountService.java

```diff
@@ -71,32 +73,32 @@
      * @return the number of roles matching the filter by each domain
      */
     public PairDTO[] countRoles(String filter) throws UserStoreCounterException {
+
         Set<String> userStoreDomains = UserStoreCountUtils.getCountEnabledUserStores();
-        //add 3 more for the counts of Internal, Application domains
+        //Add 2 more for the counts of Internal, Application domains.
         PairDTO[] roleCounts = new PairDTO[userStoreDomains.size() + 2];
         int i = 0;
 
         for (String userStoreDomain : userStoreDomains) {
-            UserStoreCountRetriever counter = UserStoreCountUtils.getCounterInstanceForDomain(userStoreDomain);
-            Long count = Long.valueOf(-1);
-            if (counter != null) {
-                try {
-                    count = counter.countRoles(filter);
-                } catch (UserStoreCounterException e) {
-                    log.error("Error while getting role count from user store domain : " + userStoreDomain, e);
-                }
-            } else {
-                //no action
+            long count = -1L;
+            String filterWithDomain = getFilterWithDomain(userStoreDomain, filter);
+
+            try {
+                count = getRoleCount(filterWithDomain);
+            } catch (UserStoreCounterException e) {
+                log.error("Error while getting role count from user store domain : " + userStoreDomain, e);
```

Comment:
As the `UserStoreCounterException` is created within this component, we do not need to catch that here. Therefore we can omit the try-catch block.